### PR TITLE
fix(embedding): omit default OpenAI dimensions

### LINKED ISSUE #1831: [Bug]: Model "Qwen3-Embedding-0.6B" does not support matryoshka representation,
### Prerequisites

- [x] I have searched the existing [issues](https://github.com/agentscope-ai/agentscope/issues) and [discussions](https://github.com/agentscope-ai/agentscope/discussions), and this is not a duplicate.
- [x] This is a bug, not a usage question. (For questions, please use [Discussions](https://github.com/agentscope-ai/agentscope/discussions/new?category=general) instead.)

### Background / Description

Error code: 400 - {'object': 'error', 'message': 'Model "Qwen3-Embedding-0.6B" does not support matryoshka representation, changing output dimensions will lead to poor results.', 'type': 'BadRequestError', 'param': None, 'code': 400}
ERROR:    Traceback (most recent call last):
  File "/home/zhangkailin/.conda/envs/ascope/lib/python3.12/site-packages/starlette/routing.py", line 694, in lifespan
    async with self.lifespan_context(app) as maybe_state:
  File "/home/zhangkailin/.conda/envs/ascope/lib/python3.12/site-packages/fastapi/routing.py", line 241, in __aenter__
    await self._router._startup()
  File "/home/zhangkailin/.conda/envs/ascope/lib/python3.12/site-packages/fastapi/routing.py", line 4884, in _startup
    await handler()
  File "/appdata/zhangkailin/agent/agentscope-1.0.21/examples/functionality/rag/service_yolo_rag_mil_v2.py", line 167, in startup_event
    await init_knowledge()
  File "/appdata/zhangkailin/agent/agentscope-1.0.21/examples/functionality/rag/service_yolo_rag_mil_v2.py", line 148, in init_knowledge
    await knowledge.add_documents(docs1)
  File "/appdata/zhangkailin/agent/agentscope-1.0.21/src/agentscope/rag/_simple_knowledge.py", line 77, in add_documents
    res_embeddings = await self.embedding_model(
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/appdata/zhangkailin/agent/agentscope-1.0.21/src/agentscope/embedding/_openai_embedding.py", line 94, in __call__
    response = await self.client.embeddings.create(**kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/zhangkailin/.conda/envs/ascope/lib/python3.12/site-packages/openai/resources/embeddings.py", line 259, in create
    return await self._post(
           ^^^^^^^^^^^^^^^^^
  File "/home/zhangkailin/.conda/envs/ascope/lib/python3.12/site-packages/openai/_base_client.py", line 1884, in post
    return await self.request(cast_to, opts, stream=stream, stream_cls=stream_cls)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/zhangkailin/.conda/envs/ascope/lib/python3.12/site-packages/openai/_base_client.py", line 1669, in request
    raise self._make_status_error_from_response(err.response) from None
openai.BadRequestError: Error code: 400 - {'object': 'error', 'message': 'Model "Qwen3-Embedding-0.6B" does not support matryoshka representation, changing output dimensions will lead to poor results.', 'type': 'BadRequestError', 'param': None, 'code': 400}

ERROR:    Application startup failed. Exiting.

### Error Messages

```shell
Model "Qwen3-Embedding-0.6B" does not support matryoshka representation,
```

### Steps to Reproduce

Error code: 400 - {'object': 'error', 'message': 'Model "Qwen3-Embedding-0.6B" does not support matryoshka representation, changing output dimensions will lead to poor results.', 'type': 'BadRequestError', 'param': None, 'code': 400}
ERROR:    Traceback (most recent call last):
  File "/home/zhangkailin/.conda/envs/ascope/lib/python3.12/site-packages/starlette/routing.py", line 694, in lifespan
    async with self.lifespan_context(app) as maybe_state:
  File "/home/zhangkailin/.conda/envs/ascope/lib/python3.12/site-packages/fastapi/routing.py", line 241, in __aenter__
    await self._router._startup()
  File "/home/zhangkailin/.conda/envs/ascope/lib/python3.12/site-packages/fastapi/routing.py", line 4884, in _startup
    await handler()
  File "/appdata/zhangkailin/agent/agentscope-1.0.21/examples/functionality/rag/service_yolo_rag_mil_v2.py", line 167, in startup_event
    await init_knowledge()
  File "/appdata/zhangkailin/agent/agentscope-1.0.21/examples/functionality/rag/service_yolo_rag_mil_v2.py", line 148, in init_knowledge
    await knowledge.add_documents(docs1)
  File "/appdata/zhangkailin/agent/agentscope-1.0.21/src/agentscope/rag/_simple_knowledge.py", line 77, in add_documents
    res_embeddings = await self.embedding_model(
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/appdata/zhangkailin/agent/agentscope-1.0.21/src/agentscope/embedding/_openai_embedding.py", line 94, in __call__
    response = await self.client.embeddings.create(**kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/zhangkailin/.conda/envs/ascope/lib/python3.12/site-packages/openai/resources/embeddings.py", line 259, in create
    return await self._post(
           ^^^^^^^^^^^^^^^^^
  File "/home/zhangkailin/.conda/envs/ascope/lib/python3.12/site-packages/openai/_base_client.py", line 1884, in post
    return await self.request(cast_to, opts, stream=stream, stream_cls=stream_cls)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/zhangkailin/.conda/envs/ascope/lib/python3.12/site-packages/openai/_base_client.py", line 1669, in request
    raise self._make_status_error_from_response(err.response) from None
openai.BadRequestError: Error code: 400 - {'object': 'error', 'message': 'Model "Qwen3-Embedding-0.6B" does not support matryoshka representation, changing output dimensions will lead to poor results.', 'type': 'BadRequestError', 'param': None, 'code': 400}

ERROR:    Application startup failed. Exiting.

### Environment

- AgentScope Version:
- Python Version:
- OS:

### PR DESCRIPTION
**Pull request by Premsenareddy**
## Summary
- make OpenAI embedding dimensions optional at runtime so OpenAI-compatible providers can use their native defaults
- only send the dimensions parameter when explicitly configured
- keep YAML model-card defaults rendering as concrete integer schemas for built-in OpenAI models

Fixes #1831

## Tests
- .venv/bin/python -m pytest tests/embedding_openai_test.py -q
- .venv/bin/python -m pytest tests/embedding_openai_test.py tests/embedding_dashscope_test.py tests/embedding_gemini_test.py tests/embedding_ollama_test.py -q -p no:cacheprovider
- .venv/bin/pre-commit run --files src/agentscope/embedding/_embedding_base.py src/agentscope/embedding/_embedding_model_card.py src/agentscope/embedding/_openai/_model.py tests/embedding_openai_test.py

### CODE REVIEW

**Comment by Premsenareddy** [comment]
Refreshed this PR against latest `main` and reshaped the fix around the current embedding config model.\n\nThe updated approach keeps `dimensions` as the first-class concrete vector size for storage/RAG collection sizing, and adds OpenAI-specific `parameters.pass_dimensions` for controlling whether the API payload includes `dimensions`. That means OpenAI-compatible providers like Qwen can persist a known vector dimension while avoiding the provider-rejected `dimensions` request parameter. Direct Python construction with `pass_dimensions=False` still works too.\n\nValidation after the refresh:\n- `tests/embedding_openai_test.py`: 8 passed\n- embedding provider slice: 29 passed\n- file-scoped pre-commit passed\n\nGitHub now reports the PR as review-blocked rather than dirty; refreshed CI is running on `8cb67298`.

File: src/agentscope/embedding/_openai/_model.py [hunk 1]
```diff
@@ -97,7 +113,11 @@ def __init__(
             api_key=credential.api_key.get_secret_value(),
             **client_kwargs,
         )
-        self.pass_dimensions = pass_dimensions
+        self.pass_dimensions = (
+            parameters.pass_dimensions
+            if parameters is not None
```
Comment (gnanirahulnutakki) [inline src/agentscope/embedding/_openai/_model.py]: [P2] Preserve the explicit constructor flag when parameters use defaults

`Parameters.pass_dimensions` defaults to `True`, so merely supplying the pre-existing `parameters` argument now overrides an explicit `pass_dimensions=False`. On this exact head, `OpenAIEmbeddingModel(..., parameters=OpenAIEmbeddingModel.Parameters(), pass_dimensions=False)` sends `dimensions=1024`; before this PR the constructor flag omitted it. That reintroduces the provider 400 this change is meant to avoid for wrappers that always instantiate `Parameters`. Please let the parameter value win only when that field was explicitly set (for example via `parameters.model_fields_set`) and add a combined-arguments regression.

File: src/agentscope/embedding/_embedding_model_card.py [hunk 2]
```diff
@@ -151,6 +151,30 @@ def from_yaml(
 
                 # Simple dict merge
                 if param_name in properties:
+                    property_schema = properties[param_name]
+                    if (
+                        override.get("default") is not None
+                        and isinstance(property_schema, dict)
+                        and isinstance(property_schema.get("anyOf"), list)
+                    ):
+                        non_null_schema = next(
```
Comment (gnanirahulnutakki) [inline src/agentscope/embedding/_embedding_model_card.py]: [P2] Preserve genuine multi-type `anyOf` schemas

This selects the first non-null branch for every `anyOf`, not only an `Optional[T]`. With `value: int | str | None` plus a YAML default of `7`, `from_yaml()` emits only `{"type": "integer", ...}` and silently drops the valid string branch; the previous simple merge preserved the union. Please flatten only when there is an explicit null branch and exactly one non-null branch, and retain multi-branch unions with regression coverage.

**Review by gnanirahulnutakki — CHANGES_REQUESTED** [review]
I reproduced both findings on head `ce2832e`. The existing focused suite (8 tests), full embedding-provider slice (29 tests after installing model extras), and all six GitHub checks pass, but the two compatibility/schema edge cases are not covered. I also validated the narrow guards described inline against the provider slice and file-scoped pre-commit hooks.

**Comment by Premsenareddy** [comment]
Addressed the two review comments in the latest push.

Changes:
- OpenAIEmbeddingModel now lets parameters.pass_dimensions win only when that field was explicitly set, so the combined Parameters() plus pass_dimensions=false constructor case still omits the API dimensions payload.
- EmbeddingModelCard.from_yaml() now flattens anyOf only for true Optional[T] schemas with one non-null branch plus null, preserving multi-branch unions such as int | str | None.

Regression coverage added:
- combined constructor/default-parameters pass_dimensions=false case
- explicit parameter override still winning over constructor flag
- optional single-branch schema flattening
- multi-branch anyOf preservation

Local validation:
- tests/embedding_openai_test.py tests/embedding_model_card_test.py: 12 passed
- embedding provider slice including DashScope/Gemini/Ollama: 33 passed
- file-scoped pre-commit: passed

**Review by gnanirahulnutakki — APPROVED** [review]
Re-reviewed exact head de155d7e934278f5538a8cbeb7e9ebf408e4b290. Both prior findings are fixed: an otherwise-default Parameters value now defers to the constructor pass_dimensions setting while an explicitly supplied parameter still takes precedence, and anyOf flattening is limited to one non-null branch plus null so genuine multi-type unions remain intact. Focused tests passed 12/12, the full embedding-provider slice passed 33/33, file-scoped pre-commit hooks passed, and all six hosted checks are green. No blocking findings remain.
